plugins/inputs/couchbase/couchbase.go

```diff
@@ -379,13 +391,39 @@ func (cb *Couchbase) Init() error {
 
 	cb.bucketInclude = f
 
+	tlsConfig, err := cb.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	cb.client = &http.Client{
+		Timeout: 10 * time.Second,
+		Transport: &http.Transport{
+			MaxIdleConnsPerHost: couchbaseClient.MaxIdleConnsPerHost,
+			TLSClientConfig:     tlsConfig,
+		},
+	}
+
+	// Couchbase requires that a TLS cert and key be provided when insecure skip verify is disabled
+	if !cb.ClientConfig.InsecureSkipVerify && (cb.ClientConfig.TLSCert == "" || cb.ClientConfig.TLSKey == "") {
+		return fmt.Errorf("tls_cert and tls_key are required when insecure_skip_verify = false")
+	}
```

Comment:
can you provide a reference to this? I don't understand this requirement. It seems like if there's no key or cert then TLS configuration should not be enabled.

Reply:
I misread some of the code in go-couchbase, the current implementation is incorrect.